File: backend/common/logger.py
import os
from datetime import datetime

class Logger(object):
    LOG_FILE_TEMPLATE = '%s.log'
    LOG_TEMPLATE = '{"component": "%s", "section": "%s", "timestamp": "%s", "message": "%s"}'
    
    def __init__(self, component):
        self.component = component
        self.log_file = self.LOG_FILE_TEMPLATE % component
    
    def log(self, section, message):
        time = datetime.utcnow().strftime('%Y-%m-%d %H:%M:%SZ')
        log = self.LOG_TEMPLATE % (self.component, section, time, message)
        with open(self.log_file, 'aw') as f:
            f.write(log.replace('\n', '') + '\n')

File: frontend/web-server/logger/logger.js
const fs = require('fs');

const log_file = 'webserver.log';

function log(section, message) {
    const timestamp = new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '');
    let log = `{"component": "webserver", "section":"${section}", "timestamp":"${timestamp}"Z, "message":"${message}"}\n`
    fs.appendFile(log_file, 
        log.replace(/\n/g, ''),
        function(err) {
            if (err) {
                console.log(err);
            }
        });
}

module.exports = {
    log: log
};